ocaml/xapi/message_forwarding.ml

```diff
@@ -2605,6 +2605,33 @@ module Forward = functor(Local: Custom_actions.CUSTOM_ACTIONS) -> struct
         (fun session_id rpc ->
            Client.Host.get_server_certificate rpc session_id host)
 
+    let install_server_certificate ~__context ~host ~certificate ~private_key ~certificate_chain =
+      let task = Context.get_task_id __context in
+      let local_fn =
+        Local.Host.install_server_certificate
+          ~host
+          ~certificate
+          ~private_key
+          ~certificate_chain
+      in
+      let success () =
+        let progress = Db.Task.get_progress ~__context ~self:task in
```

Comment:
i believe ``let success () = Task.get_status ~__context ~self:task = `success`` would be preferred

(for example migrate_send's progress gets set to 1 and then back to zero again a couple of times... :/)

Reply:
Actually, I'm wrong here - setting the task status to ```success`` would make the client think that the task had completed, when it in fact hasn't

Reply:
I used this scheme because I saw it was being used elsewhere. The way it works is: I set the progress 1 one just before restarting stunnel. As soon as it reconnect checks that the task completion is set to 1 and end successfully. If for any reason the task fails beforehand and the connection is reset this check will fail.

I used this to follow convention, I'm more than happy to change it to something else if it's better.

Reply:
if it works, and since the same technique is used elsewhere, i would leave it as it is


the only status states are:

```
  | `pending 
  | `success 
  | `failure 
  | `cancelling
  | `cancelled
```

as stated above we can't use `` `success ``,  so we can't use the task status, and there are no other valid candidates